### PR TITLE
add contributing guide for sig-node

### PR DESCRIPTION
**Pull request by EmilyM1**
Adding contributing.md guide for sig-node group.

### CODE REVIEW

File: sig-node/CONTRIBUTING.md
Comment (ehashman) [inline sig-node/CONTRIBUTING.md:21]: Link? Maybe also include a bit more detail in each of these tasks.

File: sig-node/CONTRIBUTING.md
Comment (ehashman) [inline sig-node/CONTRIBUTING.md:18]:
```suggestion
## Getting Started
```

File: sig-node/CONTRIBUTING.md
Comment (ehashman) [inline sig-node/CONTRIBUTING.md:26]:
```suggestion
```

Bazel is gone.

File: sig-node/CONTRIBUTING.md
Comment (ehashman) [inline sig-node/CONTRIBUTING.md:4]: If we're using @dims's guide https://hackmd.io/@XYdYH0X5SYC3DUYFF5Wylg/SyYoo-U1w as a starting point, there's a lot of missing links we could include in here.

I also recommend we include links to the Node Triage documentation.

**Review by ehashman — CHANGES_REQUESTED** [review]
/cc @adisky @dims

File: sig-node/CONTRIBUTING.md
Comment (adisky) [inline sig-node/CONTRIBUTING.md:1]:
```suggestion
# Contributing to SIG Node
```

File: sig-node/CONTRIBUTING.md
Comment (adisky) [reply sig-node/CONTRIBUTING.md:4]: +1, 
Also add a link to developer docs https://github.com/kubernetes/community/tree/master/contributors/devel/sig-node. It has lots of docs specific to SIG Node.

File: sig-node/CONTRIBUTING.md
Comment (adisky) [reply sig-node/CONTRIBUTING.md:21]: Either we need to link here to [Building Kubernetes](https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#building-kubernetes) or add more details in each step.

File: sig-node/CONTRIBUTING.md
Comment (adisky) [inline sig-node/CONTRIBUTING.md:23]: Link for unit tests
https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#unit-tests

File: sig-node/CONTRIBUTING.md
Comment (adisky) [inline sig-node/CONTRIBUTING.md:27]: dependency management https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/vendor.md

File: sig-node/CONTRIBUTING.md
Comment (adisky) [inline sig-node/CONTRIBUTING.md:30]: Running a local cluster
https://github.com/kubernetes/community/blob/master/contributors/devel/running-locally.md

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:21]: Makefile is pretty easy to find, went with the more thorough Building Kubernetes link.

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:18]: Changed

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:26]: It is!

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:4]: Added more

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:1]: Fixed

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:23]: Added

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:30]: Added

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:27]: Yeah, this is helpful

File: sig-node/CONTRIBUTING.md
Comment (ehashman) [inline sig-node/CONTRIBUTING.md:5]: A bunch of links are provided here with no context. I think it would be useful to explain why each of these are helpful. A new contributor is not going to know what enhancements are or why they are relevant to SIG Node.

File: sig-node/CONTRIBUTING.md
Comment (ehashman) [inline sig-node/CONTRIBUTING.md:32]: The indentation on this section is causing it to render as code. The links aren't currently clickable.

File: sig-node/CONTRIBUTING.md
Comment (ehashman) [inline sig-node/CONTRIBUTING.md:38]: Can we please add links?

File: sig-node/CONTRIBUTING.md
Comment (SergeyKanzhelev) [inline sig-node/CONTRIBUTING.md:15]: as minimum, probe folder is also worth mentioning:

kubernetes/pkg/probe/
kubernetes/vendor/k8s.io/kubelet/
kubernetes/vendor/k8s.io/cri-api

File: sig-node/CONTRIBUTING.md
Comment (SergeyKanzhelev) [inline sig-node/CONTRIBUTING.md:32]: is it node specific?

Comment (ehashman) [reply sig-node/CONTRIBUTING.md:32]: Yes, this is Node's onboarding guide as opposed to project-wide.

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:5]: The heading above each link explains.

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:32]: Fixed

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:38]: Added

**Comment by ehashman** [comment]
also
/cc @dims 

as this was based on your guide in hackmd :)

File: sig-node/CONTRIBUTING.md
Comment (EmilyM1) [reply sig-node/CONTRIBUTING.md:15]: Added

**Comment by EmilyM1** [comment]
/assign @derekwaynecarr
/assign @dchen1107


Can this merge?

**Comment by SergeyKanzhelev** [comment]
> Can this merge?

it doesn't render right. Can you please fix: https://github.com/kubernetes/community/blob/2ad5cebfc3b5c7aa7a5d1bf3c3d7cd31bbd58e98/sig-node/CONTRIBUTING.md

to see how it renders you can look at file in your branch or click here:
image

**Comment by EmilyM1** [comment]
@SergeyKanzhelev Thanks for the comment! Should be all fixed here:
https://github.com/kubernetes/community/blob/8086c41d0d9ab1695d3b19ddd38b8ef0a584bc48/sig-node/CONTRIBUTING.md

**Comment by SergeyKanzhelev** [comment]
as the very first iteration, 

/lgtm

perhaps we can iterate on it with smaller PRs?

If we want a finished contributing.md, we may want to start a google doc for collaborative contributions.

**Comment by EmilyM1** [comment]
@SergeyKanzhelev I was asked to throw this together at a Reviewer meeting. This is based off a doc from @dims . I am fine with any adds to this. I think it would be a good idea to get this merged and then folks could mod it. The thinking there is- some contributor doc is better than our current no contributor doc.

**Comment by EmilyM1** [comment]
@dims @ehashman  can this get lgtm/approve/merge ?

**Comment by ehashman** [comment]
Neither of us have approver, perhaps @dchen1107 can TAL.

/assign @dchen1107

**Comment by EmilyM1** [comment]
/assign @derekwaynecarr
/assign @dchen1107

can this get /lgtm /approve merge?

**Comment by dchen1107** [comment]
/lgtm
/approve

Emily and Sergey, thanks for this. All are great stuff!

**Pull request merged**
